File: framework/languages/dotnet/samples/DeliveryDispatch/Server/Configuration/SampleNames.cs
namespace DeliveryDispatch.Server.Configuration;

public static class SampleNames
{
    public const string CourierRouteChannel = "deliverydispatch.courier";
    public const string CourierSpotRouteChannel = "delivery-couriers.route";
    public const string TrackingRouteChannel = "deliverydispatch.tracking";
    public const string CustomerRouteChannel = "deliverydispatch.customer";
    public const string CustomerActorDiscovery = "delivery-customers";
    public const string CourierActorDiscovery = "delivery-couriers";
    public const string CustomerSpotNode = "delivery-customer-node";
    public const string CourierSessionSpotNode = "delivery-courier-session-node";
    public const string CourierSpotNode1 = "delivery-courier-node-1";
    public const string CourierSpotNode2 = "delivery-courier-node-2";
    public const string CourierEntrySpotNode1 = "delivery-courier-entry-1";
    public const string CourierEntrySpotNode2 = "delivery-courier-entry-2";
    public const string CustomerStreamNode = "delivery-customer-stream";
    public const string CourierStreamNode = "delivery-courier-stream";
    public const string CustomerActorType = "delivery-customer";
    public const string CourierActorType = "delivery-courier";
}

public static class SampleTimings
{
    public static readonly TimeSpan FrameworkTimeout = TimeSpan.FromSeconds(5);
    public static readonly TimeSpan DispatchTimeout = TimeSpan.FromMilliseconds(700);
    public static readonly TimeSpan ClientTimeout = TimeSpan.FromSeconds(12);
}

File: framework/languages/node/samples/DeliveryDispatch.Ts/Shared/Configuration/sample-names.ts
const SampleNames = {
  dispatchChannel: 'deliverydispatch.dispatch',
  routeMesh: 'deliverydispatch.mesh',
  courierStreamNode: 'delivery-courier-stream',
  trackingChannel: 'deliverydispatch.tracking',
  customerStreamNode: 'delivery-customer-stream',
  courierActorType: 'delivery-courier',
  customerActorType: 'delivery-customer'
} as const;

const SampleTimings = {
  requestTimeout: 10000,
  offerDecisionTimeout: 700,
  offerSweepInterval: 50,
  clientTimeout: 30000
} as const;

function courierActorNodeRid(courierId: string): string {
  if (courierId === 'courier-a') {
    return 'courier-node-1';
  }
  if (courierId === 'courier-b') {
    return 'courier-node-2';
  }
  throw new Error(`Unknown courier '${courierId}'.`);
}

export {
  SampleNames,
  SampleTimings,
  courierActorNodeRid
};
